src/cert_context.rs

```diff
@@ -107,6 +107,54 @@ impl CertContext {
         self.get_encoded_bytes()
     }
 
+    /// Deletes the cert's associated key container for both CAPI and CNG keys
+    pub fn del_key_container(&mut self) -> io::Result<()>{
+        unsafe {
+            // get CERT_KEY_PROV_INFO_PROP_ID and cast to key_prov
+            let bytes = self.get_bytes(Cryptography::CERT_KEY_PROV_INFO_PROP_ID)?;
+            assert!(bytes.len() <= u32::max_value() as usize);
+            let key_prov = bytes.as_ptr() as *const Cryptography::CRYPT_KEY_PROV_INFO;
+            
+            if (*key_prov).dwKeySpec == 0 { // CNG Key
+                let mut prov_handle = Cryptography::NCRYPT_PROV_HANDLE::default();
```

Comment:
>let mut prov_handle [](http://example.com/codeflow?start=16&length=19)

Will  NCryptFreeObject be called when this dropped?

Reply:
For any C returned objects, you will need to explicitly free or store in a Rust wrapper object that supports a drop.

See pub struct CertStore(Cryptography::HCERTSTORE); and its drop.